x/launch/keeper/msg_server_settle_request_test.go

```diff
@@ -0,0 +1,140 @@
+		name      string
+		msg       types.MsgSettleRequest
+		checkAddr string
+		err       error
+	}{
+		{
+			name: "invalid chain",
+			msg: types.MsgSettleRequest{
+				ChainID:     invalidChain,
+				Coordinator: coordinator1.Address,
+				RequestID:   requests[0].RequestID,
+				Approve:     true,
+			},
+			err: sdkerrors.Wrap(types.ErrChainNotFound, invalidChain),
+		}, {
+			name: "launch triggered chain",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[0].ChainID,
+				Coordinator: coordinator1.Address,
+				RequestID:   requests[0].RequestID,
+				Approve:     true,
+			},
+			err: sdkerrors.Wrap(types.ErrTriggeredLaunch, chains[0].ChainID),
+		}, {
+			name: "no permission error",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: coordinator2.Address,
+				RequestID:   requests[0].RequestID,
+				Approve:     true,
+			},
+			err: sdkerrors.Wrap(types.ErrNoAddressPermission, coordinator2.Address),
+		}, {
+			name: "settle a invalid request",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID),
+				RequestID:   99999999,
+				Approve:     true,
+			},
+			err: sdkerrors.Wrapf(types.ErrRequestNotFound,
+				"request 99999999 for chain %s not found", chains[1].ChainID),
+		}, {
+			name: "add chain 1 request 1",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID),
+				RequestID:   requests[0].RequestID,
+				Approve:     true,
+			},
+			checkAddr: addr1,
+		}, {
+			name: "add chain 1 request 2",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID),
+				RequestID:   requests[1].RequestID,
+				Approve:     true,
+			},
+			checkAddr: addr2,
+		}, {
+			name: "add chain 1 request 3",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID),
+				RequestID:   requests[2].RequestID,
+				Approve:     true,
+			},
+			checkAddr: addr3,
+		}, {
+			name: "add chain 2 request 4",
+			msg: types.MsgSettleRequest{
+				ChainID:     chains[1].ChainID,
+				Coordinator: pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID),
+				RequestID:   requests[3].RequestID,
+				Approve:     true,
+			},
+			checkAddr: addr4,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := srv.SettleRequest(ctx, &tt.msg)
+			if tt.err != nil {
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+
+			_, found := k.GetRequest(sdkCtx, tt.msg.ChainID, tt.msg.RequestID)
+			require.False(t, found, "request not removed")
+
+			_, found = k.GetGenesisAccount(sdkCtx, tt.msg.ChainID, tt.checkAddr)
+			require.True(t, found, "request apply not found")
```

Comment:
We should assert it exists when `msg.Approve == true` and it doesn't when `msg.Approve == false`